### ISSUE TITLE
Exam bank: Fix term name of pre-1999 exams

### ISSUE THREAD

**Issue by some-molasses** [MathSoc/mathsoc-website#316]
![image](https://github.com/user-attachments/assets/ea45c5f4-8eb7-49eb-b4f2-c2c0e072d149)

Looks like some exams aren't correctly being mapped to their associated term name. It's only some of them though - we should try to figure out what's going on, and fix it.

**Comment by sai3000pro** [comment]
Not too sure how to add myself to this issue but I'd be happy to try and investigate this!

**Comment by sai3000pro** [comment]
Just curious, do we know that pre-1999 exams all have their term code instead of "season year"? I can just check if the term is an integer then and apply the conversion... I see that 1245=Spring 2024, 1249=Fall 2024, 1251=Winter 2025, 1255 = Spring 2025. Was this pattern upheld for older terms?

**Comment by some-molasses** [comment]
you can see on the [current exam bank](https://services.mathsoc.uwaterloo.ca/resources/exam-bank) that it's broken - although now it's broken for all exams ;-;

The pattern was upheld for older terms! For any term code `abcd`, Waterloo maps it to year `2000+abc`, month `d` - that's been consistent since the 90s at least, probably before that too?

**Comment by sai3000pro** [comment]
;-;

So I'm assuming that "a" tells you if it was before or after 2000? and bc would be the year, d is the month?

**Comment by some-molasses** [comment]
yeah pretty much - `abc` is just a 2-3 digit number, in years pre-2000, the term code might just be something like "995" (with no `a` digit)

**Comment by sai3000pro** [comment]
Also is there a way to test this without pushing? Since for me on localhost I can't seem to see any exams at all

![Image](https://github.com/user-attachments/assets/f0e73537-94aa-4151-9f09-c6ba7c31bf3d)

**Comment by some-molasses** [comment]
responded on discord - you'll just need to upload a few exams locally

**Comment by sai3000pro** [comment]
One more thing - for code maintainability purposes, would the year 2100 lead to term codes starting with 2? Not necessary for me to complete but just wanted to ask in case you knew

**Comment by some-molasses** [comment]
lol this probably falls firmly into the realm of other people's problem, but yeah might as well implement it if it's no more difficult
